source/geometry_model/interface.cc

```diff
@@ -36,6 +36,22 @@ namespace aspect
     {}
 
 
+    template <int dim>
+    void
+    Interface<dim>::update ()
+    {}
+
+
+    template <int dim>
+    double
+    Interface<dim>::depth_including_mesh_deformation(const Point<dim> &) const
+    {
+      AssertThrow(false,
+                  ExcMessage("The depth including mesh deformation function only works with a 2D box geometry model."));
```

Comment:
Better to write this message generically. Once we implement this for a few more models the current form will become out of date.
```suggestion
                  ExcMessage("The function depth_including_mesh_deformation () was called, but it is not implemented for the current geometry model."));
```